### PR TITLE
Task/tts for play

### PR DESCRIPTION
**Pull request by paufabregat**
TTS is executed for all types user wants to play. Error case is also handled.

### CODE REVIEW

File: snipssonos/entities/album.py [hunk 1]
```diff
@@ -2,15 +2,17 @@
 
 
 class Album(object):
-    def __init__(self, uri, name):
+    def __init__(self, uri, name, artist_name=None):
         self.uri = uri
         self.name = name
+        self.artist_name = artist_name
```
Comment (anthonyray) [inline snipssonos/entities/album.py]: why not attach an optional artist entity instead ?

File: snipssonos/entities/track.py [hunk 2]
```diff
@@ -2,18 +2,23 @@
 
 
 class Track(object):
-    def __init__(self, uri, name=None):
+    def __init__(self, uri, name=None, artist_name=None):
         self.uri = uri
         self.name = name
+        self.artist_name = artist_name
```
Comment (anthonyray) [inline snipssonos/entities/track.py]: Same here

File: snipssonos/services/spotify/music_search_service.py [hunk 3]
```diff
@@ -209,8 +210,16 @@ def _parse_album_results(self, raw_response):
         response = json.loads(raw_response)
         albums = response['albums']
 
-        artists = [Album(item['uri'], item['name']) for item in albums['items']]
-        return artists
+        albums = [Album(item['uri'], item['name'], self._get_artists_name_from_item(item))
+                  for item in albums['items']]
+        return albums
+
+    def _get_artists_name_from_item(self, item):
```
Comment (anthonyray) [inline snipssonos/services/spotify/music_search_service.py]: We could parse it as artists.

File: snipssonos/use_cases/play/artist.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 from snipssonos.shared.use_case import UseCase
 from snipssonos.shared.response_object import ResponseSuccess, ResponseFailure
 
-from snipssonos.shared.feedback import FR_TTS_GENERIC_ERROR  # TODO : This might violate the dependency rule of clean Architecture ...
+from snipssonos.shared.feedback import FR_TTS_GENERIC_ERROR, FR_TTS_PLAYING_ARTIST_TEMPLATE  # TODO : This might violate the dependency rule of clean Architecture ...
```
Comment (anthonyray) [inline snipssonos/use_cases/play/artist.py]: I wonder if we should not move this logic to a Feedback component / Feedback service.

File: snipssonos/use_cases/play/playlist.py [hunk 5]
```diff
@@ -13,15 +13,17 @@ def __init__(self, device_discovery_service, music_search_service, music_playbac
...
-            else:
-                return ResponseFailure.build_resource_error(FR_TTS_GENERIC_ERROR)
-
-        return ResponseSuccess(feedback=tts_feedback)
+
+        if len(results_playlists):
+            first_playlist = results_playlists[0]
+            self.music_playback_service.clear_queue(device)
+            self.music_playback_service.play(device, first_playlist)
+            tts_feedback = FR_TTS_PLAYING_PLAYLIST_TEMPLATE.format(first_playlist.name)
```
Comment (anthonyray) [inline snipssonos/use_cases/play/playlist.py]: Yeah, the more I think about it, the more I would to see a `feedback` service.

**Review by anthonyray — CHANGES_REQUESTED** [review]
Refactor the TTS feedback into a dedicated service. 
The service should provide APIs to easily switch the language.

File: action-sonos.py [hunk 6]
```diff
@@ -286,6 +288,7 @@ def playMusic_callback(hermes, intentMessage):
         h.device_transport_control_service = NodeDeviceTransportControlService(CONFIGURATION)
         h.music_search_service = SpotifyMusicSearchService(client_id, client_secret, refresh_token)
         h.music_playback_service = NodeMusicPlaybackService(CONFIGURATION=CONFIGURATION)
+        h.feedback_service = FeedbackService(LANGUAGE)
```
Comment (anthonyray) [inline action-sonos.py:291]: 😍

File: snipssonos/services/feedback/feedback_service.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        }
+    }
+
+    SUPPORTED_LANGUAGES = ['en', 'fr']
+
+    def __init__(self, language):
+        self.validate_language(language)
+        self.language = language
+
+    def set_language(self, language):
```
Comment (anthonyray) [inline snipssonos/services/feedback/feedback_service.py:37]: You can use setters here.

File: snipssonos/services/feedback/feedback_service.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        self.validate_language(language)
+        self.language = language
+
+    def get_generic_error_message(self):
```
Comment (anthonyray) [inline snipssonos/services/feedback/feedback_service.py:41]: Same here, it would be cleaner to have getter properties.

File: snipssonos/entities/album.py [hunk 9]
```diff
@@ -2,15 +2,17 @@
 
 
 class Album(object):
-    def __init__(self, uri, name):
+    def __init__(self, uri, name, artist_name=None):
```
Comment (anthonyray) [inline snipssonos/entities/album.py]: Still no artist entity 😢 ?

File: snipssonos/services/feedback/feedback_service.py [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    def get_artist_template(self):
+        return self.FEEDBACK_OBJECT[self.language]['artist']
+
+    def get_track_info_template(self):
+        return self.FEEDBACK_OBJECT[self.language]['track_info']
+
+    def get_no_tracks_error_message(self):
+        return self.FEEDBACK_OBJECT[self.language]['no_tracks_error']
+
+    def validate_language(self, language):
```
Comment (anthonyray) [inline snipssonos/services/feedback/feedback_service.py]: nice

File: snipssonos/services/feedback/feedback_service.py [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        if not (language in self.SUPPORTED_LANGUAGES):
+            class_name = self.__class__.__name__
+            raise AttributeError(
+                "Tried to assign an unsupported language to the language property of {}".format(class_name))
+        return True
+
+    def from_response_object(self, response_object):
+        if response_object:  # ResponseSuccess
+            return response_object.feedback
```
Comment (anthonyray) [inline snipssonos/services/feedback/feedback_service.py:82]: Not sure we should use the feedback attribute here. Don't we have many methods to provide feedback available ?

**Review by anthonyray — CHANGES_REQUESTED** [review]

File: snipssonos/services/feedback/feedback_service.py (hunk 11 above)
Comment (paufabregat) [reply snipssonos/services/feedback/feedback_service.py:82]: Not sure what you mean, can you rephrase? 

PS: this is actually your code that I picked form the TTSGenerator which I merged with the functionalities of the this service.

File: snipssonos/entities/album.py (hunk 9 above)
Comment (paufabregat) [reply snipssonos/entities/album.py]: yes! check last commits :)

**Review by anthonyray — APPROVED** [review]

**Comment by danifcastro** [comment]
Hi,
Have you pushed these changes to the Raspi of Truth? Still not seeing the results.
Thanks,
Dani

**Comment by anthonyray** [comment]
I don't know if Pau's pushed these changes. I haven't. I will.
